### PR TITLE
Allow string argument for disable/enable/exclude

### PR DESCRIPTION
**Pull request by svlandeg**
## Description
When loading an `nlp` pipeline, we assumed and documented that `disable`, `enable` and `exclude` should be lists, but this wasn't enforced or checked, resulting in silent odd behavior when giving these arguments a single string:

```
nlp = spacy.load("en_core_web_md", disable="tok2vec")
print(nlp.pipe_names)
#  ['tok2vec', 'tagger', 'parser', 'attribute_ruler', 'lemmatizer', 'ner']
print(nlp._disabled)
# {'t', 'e', 'senter', 'c', 'v', 'o', 'k', '2'}
```

`exclude="tok2vec"` did in fact work, [by accident](https://github.com/explosion/spaCy/blob/master/spacy/language.py#L1809), because `"tok2vec" in "tok2vec"` equals `True` because of substring comparisons.

In this PR, we broaden the typing of the loading functions to include `str` as an option, and turn the string into a `list` before anything else. This makes the functionality consistent with the current implementation of `nlp.select_pipes`, which also caters for both a single string and a list. Note that the implementation of `_resolve_component_status` was allowing for `disable` to be `str` but the typing didn't reflect that, so I updated that as well.  

As part of the docs update, I also added the 3.4 tags for the new `enable` argument from https://github.com/explosion/spaCy/pull/10784

### Types of change
bug/consistency fix

## Checklist
- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by rmitsch — APPROVED** [review]

File: spacy/language.py [hunk 1]
```diff
@@ -1711,12 +1711,12 @@ def from_config(
 
         config (Dict[str, Any] / Config): The loaded config.
         vocab (Vocab): A Vocab object. If True, a vocab is created.
-        disable (Iterable[str]): Names of pipeline components to disable.
+        disable (Union[str, Iterable[str]]): Names of pipeline components to disable.
```
[the remark this answers is not in the document]
Comment (svlandeg) [reply spacy/language.py]: Thanks, I love nitpicks ;-)

File: spacy/language.py [hunk 2]
```diff
@@ -1727,6 +1727,12 @@ def from_config(
 
         DOCS: https://spacy.io/api/language#from_config
         """
+        if disable is not None and isinstance(disable, str):
+            disable = [disable]
+        if enable is not None and isinstance(enable, str):
+            enable = [enable]
+        if exclude is not None and isinstance(exclude, str):
+            exclude = [exclude]
```
[the remark this answers is not in the document]
Comment (svlandeg) [reply spacy/language.py:1735]: Yea I was a bit on the fence on this one, but I don't think `None` is ever really passed or checked in this context, so we're probably fine.

File: spacy/language.py [hunk 3]
```diff
@@ -2031,14 +2037,16 @@ def to_disk(
 
     @staticmethod
     def _resolve_component_status(
-        disable: Iterable[str], enable: Iterable[str], pipe_names: Collection[str]
+        disable: Union[str, Iterable[str]],
+        enable: Union[str, Iterable[str]],
+        pipe_names: Collection[str],
```
[the remark this answers is not in the document]
Comment (svlandeg) [reply spacy/language.py:2042]: The implementation had a check on `disable` being `str` which meant it was implictely allowing / catering for `Union[str, Iterable[str]]` so I changed the types accordingly.

This is an internal function, and only called once in the code base, and at that point a list is generated, so neither `disable` or `enable` are ever anything else than a list. We could also make the typing more strict again, and remove the `isinstance` checks, but then again it doesn't really hurt to have this additional bit of robustness?

File: spacy/language.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrianeboyd) [reply spacy/language.py]: I mean only that the `None` checks are 100% redundant here. `isinstance(x, str)` is always `False` for `None`.

File: spacy/language.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (adrianeboyd) [reply spacy/language.py:2042]: This didn't highlight like I expected. I only meant for `pipe_names`, which conflicts with the docstring below.

File: spacy/language.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (svlandeg) [reply spacy/language.py]: Oh, right! Actually, I think there's one more place in the code base where I can clean that up.

**Pull request merged**